src/compiler/scala/tools/nsc/classpath/DirectoryClassPath.scala

```diff
@@ -199,6 +218,75 @@ final class JrtClassPath(fs: java.nio.file.FileSystem) extends ClassPath with No
     dottedClassName.substring(0, dottedClassName.lastIndexOf("."))
 }
 
+/**
+  * Implementation `ClassPath` based on the $JAVA_HOME/lib/ct.sym backing http://openjdk.java.net/jeps/247
+  */
+final class CtSymClassPath(release: Option[String]) extends ClassPath with NoSourcePaths {
+  import java.nio.file.Path, java.nio.file._
+  type F = Path
+  private val javaHome = System.getProperty("java.home")
+  private val file = Paths.get(javaHome).resolve("lib").resolve("ct.sym")
```

Comment:
I've moved the check for existence out into the factory that constructs `CtSymClassPath`, but haven't escalated the error yet, as per the comment I'm adding:

```
      // TODO escalate errors once we're sure they are fatal
      // I'm hesitant to do this immediately, because -release will still work for multi-release JARs
      // even if we're running on a JRE or a non OpenJDK JDK where ct.sym is unavailable.
      //
      // Longer term we'd like an official API for this in the JDK
      // Discussion: http://mail.openjdk.java.net/pipermail/compiler-dev/2018-March/thread.html#11738
```